src/main.cpp

```diff
@@ -1,38 +1,150 @@
+                                 std::string(maxHostLength - server.host.length(), ' '),
+                                 server.username);
+    }
+}
+
+void printVersion() {
+    std::cout << fmt::format("TelemetryJet CLI (version {}.{}.{}, platform {}, architecture {})\n",
+                             CLI_VERSION_MAJOR,
+                             CLI_VERSION_MINOR,
+                             CLI_VERSION_PATCH,
+                             CLI_VERSION_SYSTEM,
+                             CLI_VERSION_ARCH);
+}
 
 int main(int argc, char** argv) {
     CLI::App app{"TelemetryJet CLI"};
 
-    auto versionCommand = app.add_subcommand("version", "Display the version and exit.")->group("Metadata");
-    auto updateCommand = app.add_subcommand("update", "Check for updates and install if available.")->group("Metadata");
-    auto serverCommand = app.add_subcommand("server", "Manage TelemetryJet server connections.")->group("Outputs");
-    auto serverListCommand = serverCommand->add_subcommand("list", "List the available server connections.")->group("Outputs");
-    auto serverClearCommand = serverCommand->add_subcommand("clear", "Clear all the server connections.")->group("Outputs");
-    auto serverAddCommand = serverCommand->add_subcommand("add", "Add a server connection.")->group("Outputs");
-    auto serverRemoveCommand = serverCommand->add_subcommand("remove", "Remove a server connection.")->group("Outputs");
+    std::string serverAlias;
+    std::string serverHost;
+    std::string serverUsername;
+    std::string serverPassword;
+    bool confirmation;
+
+    auto versionCommand
+        = app.add_subcommand("version", "Display the version and exit.")->group("Metadata");
+    auto updateCommand = app.add_subcommand("update", "Check for updates and install if available.")
+                             ->group("Metadata");
+    auto serverCommand
+        = app.add_subcommand("server", "Manage TelemetryJet server connections.")->group("Outputs");
 
-    auto streamCommand = app.add_subcommand("stream", "Connect to one or more devices.")->group("Streaming");
+    // server subcommands
+    auto serverAddCommand
+        = serverCommand->add_subcommand("add", "Add a server connection.")->group("Outputs");
+    serverAddCommand->add_option("--alias", serverAlias, "Specify an alias for the server.")
+        ->required();
+    serverAddCommand->add_option("--host", serverHost, "Specify the url of the server instance.")
+        ->required();
+    serverAddCommand->add_option("--username", serverUsername, "Specify the username.")->required();
+    serverAddCommand->add_option("--password", serverPassword, "Specify the password.")->required();
+
+    auto serverListCommand
+        = serverCommand->add_subcommand("list", "List the available server connections.")
+              ->group("Outputs");
+    auto serverRemoveCommand
+        = serverCommand->add_option("--remove", serverAlias, "Remove a server connection.")
```

Comment:
I don't think you can add an option with a text argument unless you use the --. Alternatively I could change this to:
`jet server remove --alias <alias>`